src/utils/web.py

```diff
@@ -145,7 +147,12 @@ def getUrlFd(url, headers=None, data=None, timeout=None):
             request.add_data(data)
         httpProxy = force(proxy)
         if httpProxy:
-            request.set_proxy(httpProxy, 'http')
+            proxyHandler = ProxyHandler({
+                'http': httpProxy,
+                'https': httpProxy
+                })
+            proxyOpener = build_opener(proxyHandler)
+            install_opener(proxyOpener)
```

Comment:
I'm just trying to understand your recommendation.

> That installs the proxy globally, until the next call to `getUrlFd`.

Yes. This shouldn't cause any issues since util-web is basically one of only two script files that makes direct use of `urllib`. The other file is a test script.

With `getUrlFd` being a relatively low-level method, it's called by all other request-related methods within util-web.
i.e. Any request-related method call would indirectly lead to the execution of `getUrlFd`, and thus the setting of the global proxy

> So if the config value is changed, this is not applied to plugins not using `getUrlFd`.

Correct, which is the same way it presently works without this commit.
Any plugin that does not currently make use of util-web for HTTP(S) requests will not benefit from this commit.

I believe we both agree that we should probably avoid the need for any plugins to unnecessarily configure the proxy themselves. Surely, all plugins should rather rely on the internal util-web component.

> Setting the proxy from the config itself would be better.

Just to confirm, is your main concern simply that the activation of a global HTTP(S) proxy should rather be done in within some configuration-related logic so that global proxy changes reflect immediately?
i.e. Update the global proxy as soon as the configuration option is changed